Add Docker support with Redis integration

tomascohen opened the pull request:
- Add Dockerfile with Node.js 18 Alpine base
- Add docker-compose.yml with Coce and Redis services
- Add config.docker.json for containerized environment
- Add .dockerignore for optimized builds
- Add DOCKER.md with usage documentation
- Include persistent volumes for Redis data and cached images
- Implement health checks and security best practices

fredericd commented:
Great! It looks sleek. Out of curiosity, why haven't you integrated pm2 in order to have several processes sharing the workload?

tomascohen commented:
Thanks!

I moved forward and streamlined it a bit more, making it not use a config.json but relying entirely on ENV.

I decided not to use `pm2` because it is not recommended for a couple reasons. One of them is orchestration happens at another level with containers, so we want to let the chosen orchestration tool scale the workers, etc. Second -and more important- we really want to let the container tools handle logs.

Long story short, `pm2` is the way to do for on premise deployments, and not the best for Docker.

I'm gonna overwrite this PR once I figure a solution for an outstanding small glitch.

fredericd commented:
OK. I see. And it's true that with an ultra light distribution like Alpine, and remote persistence, your docker approach should be very efficient, and harmonized in your deployment environment. I can't wait to see that...

tomascohen committed:
Add GitHub Actions workflows for CI/CD and security

- Add CI workflow for automated testing on push/PR
- Add Docker build workflow with multi-platform support
- Add security workflow with dependency scanning
- Add package-lock.json for reproducible builds
- Update .gitignore to include package-lock.json
- Add comprehensive documentation in GITHUB_ACTIONS.md
- Restore config.json.sample for upstream compatibility

Signed-off-by: Tomas Cohen Arazi <tomascohen@theke.io>

tomascohen commented:
@fredericd I'm about to submit. This is what I've done:

- Added a Dockerfile for creating the Docker image based on `node:18`.
- Added a custom entrypoint script that basically deals with translating ENV variables into the `config.json` file.
- Added basic docker-related files for documenting it and also to not copy unneeded things into the image.
- Added a sample .env file
- Added a check for security issues in libs. This I borrowed from other project and adapted. It helped me identify libraries that could be updated. I merged this into one of the commits after my check cleared all errors.
- Added a skeleton for running the tests against different node versions (WIP). this doesn't work, but doesn't block the build process. If others like it, we can work on that. Not my priority right now. I haven't even looked at the existing tests.
- A build step that publishes the image.

I also propose a workflow for new releases. From now on once you want to release, you will run:

```shell
npm version x.y.z
git push
git push origin vx.y.z
```

where `x.y.z` is your desired version. This will tweak the `package*.json` files as needed (I do that for plugins) and will also generate a commit for the release, and a tag. The last two commands will -obviously- push the assets and trigger the build pipeline.

tomascohen committed:
Add Docker support with Redis integration

- Add Dockerfile with Node.js 18 Alpine base
- Add docker-compose.yml with Coce and Redis services
- Add docker-entrypoint.sh for environment variable configuration
- Add .dockerignore for optimized builds
- Add .env.example with all configurable environment variables
- Add DOCKER.md with comprehensive usage documentation
- Include persistent volumes for Redis data and cached images
- Implement health checks and security best practices
- Preserve config.json.sample for backward compatibility

Signed-off-by: Tomas Cohen Arazi <tomascohen@theke.io>

tomascohen commented:
This is ready! I launched it as a test [here](https://coce.theke.io) using the ByWater image. I will add a sample production-ready compose file and instructions once we have an official image released here \o/

fredericd commented:
Merged! With thanks. I've pushed a v1.3.0 tag.
